src/routes/(app)/+layout.ts

```diff
@@ -78,6 +80,40 @@ export const load: LayoutLoad = async function ({
 
   const user = getAuthUser();
 
+  // hello hackness my old friend
+  // i've come to auth with you again
+  // because a session softly stale-ing
+  // left itself while devs were sleeping
+  // and the token that was planted yesterday
+  // still remains
+  // within the sound of login
+  //
+  // 🎵 to the tune of "The Sound of Silence" 🎵
+  //
+  // save the redirect and the click through the login page, iff there is an expired id token.
+  // the login page is still used to display auth errors (e.g. UI proxy)
+  if (user?.idToken) {
+    let token: JwtPayload;
+    try {
+      token = jwtDecode(user.idToken);
+    } catch (e) {
+      if (e instanceof InvalidTokenError) {
+        clearAuthUser();
+      }
+
+      throw e;
+    }
+
+    if (token?.exp && token.exp * 1000 <= Date.now()) {
+      clearAuthUser();
+      const location = new URL(document.location.toString());
+      redirect(
+        302,
+        routeForImplicitFlow(settings, location.searchParams, location.origin),
+      );
+    }
+  }
```

Comment:
So the very first time a user navigates to a Temporal UI for a new cluster/datacenter they will need to click the login button right? 
Would be nice if we could remove that click as well in the future (like how the Atlas UI/other internal web apps never require a login click) but that's a super small optimization so not high priority